src/main/kotlin/server/Telnet.kt

```diff
@@ -0,0 +1,68 @@
+package com.dlfsystems.server
+
+import com.dlfsystems.app.Log
+import io.ktor.network.selector.*
+import io.ktor.network.sockets.*
+import io.ktor.utils.io.*
+import kotlinx.coroutines.*
+
+object Telnet {
+
+    private var job: Job? = null
+
+    // Separate thread, to avoid blocking MCP on client buffer flushes (and vice versa)
+    private val scope = CoroutineScope(
+        SupervisorJob() +
+                Dispatchers.IO.limitedParallelism(1) +
+                CoroutineName("Yegg telnet")
+    )
+
+    fun start() {
+        if (job?.isActive == true) throw IllegalStateException("Already started")
+        job = Yegg.launch { server() }
+    }
+
+    fun stop() {
+        job?.cancel()
+    }
+
+    private suspend fun server() {
+        val serverSocket = aSocket(SelectorManager(Dispatchers.IO))
+            .tcp().bind(Yegg.serverAddress, Yegg.serverPort)
+        Log.i("Server listening at ${serverSocket.localAddress}:${serverSocket.port}")
+
+        while (true) {
+
+            val client = serverSocket.accept()
```

Comment:
It took me a while to realize this function is executing in the Yegg scope.  I think you probably want this stuff to be happening in the server scope, and only crossing over to the Yegg scope when it is doing Yegg stuff (see the callouts on those below).